test/simplehashregistrar_test.js

```diff
@@ -712,157 +712,45 @@ describe('SimpleHashRegistrar', function() {
+	it("releases deed after one year", function(done) {
+		var sealedBid = null;
+		var winnerBalance = 0;
 
-							assert.ok(balance.toFixed() > bidData[0].startingBalance);
-							done();
-						});
-					});
-				// });
-			},
-			// Attempt to release the deed twice
-			function(done) {
-				registrar.releaseDeed(web3.sha3('releasename'), {from: bidData[0].account}, function(err, txid) {
-					assert.notEqual(err, null, err);
-					console.log("\t Could not release deed twice");
-					done();
-				});
-			},
-			// Check the name has the correct state and owner
-			function(done) {
-				registrar.entries(web3.sha3('releasename'), function(err, result) {
-					assert.equal(err, null, err);
-					assert.equal(result[0], 0); // status == Open
-					console.log("\t Name is Open");
-					done();
-				});
-			},
-			// Check the owner is set to 0 in ENS
-			function(done) {
-				ens.owner(releasenameDotEth, function(err, owner) {
-					assert.equal(err, null, err);
-					assert.equal(owner, 0);
-					done();
-				});
-			},
-			// Check the resolver is set to 0 in ENS
-			function(done) {
-				ens.resolver(releasenameDotEth, function(err, resolver) {
-					assert.equal(err, null, err);
-					assert.equal(resolver, 0);
-					done();
-				});
-			},
-			function(done) {
-				web3.eth.getBlock('latest', function(err, block) {
-					startdate = block.timestamp;
-					done();
-				});
-			},
-			// Check we can start an auction on the name
-			function(done) {
-				registrar.startAuction(web3.sha3('releasename'), {from: bidData[0].account}, done);
-			},
-			// Check that the end time is set correctly
-			function(done) {
-				registrar.entries(web3.sha3('releasename'), function(err, result) {
-					assert.equal(err, null, err);
-					var expectedEnd = startdate + days(5);
-					assert.ok(Math.abs(result[2].toNumber() - expectedEnd) < 5, Math.abs(result[2].toNumber() - expectedEnd)); // registrationDate
-					done();
-				});
-			},
-		], done);
+		advanceTimeAsync(launchLength)
+			.then((done) => registrar.startAuctionAsync(web3.sha3('name'), {from: accounts[0]}))
+			.then((done) => registrar.shaBidAsync(web3.sha3('name'), 1e18, 1))
+			.then((result) => {
+				sealedBid = result;
+				return registrar.newBidAsync(result, {from: accounts[0], value: 1e18});
+			})
+			.then((done) => registrar.shaBidAsync(web3.sha3('name'), 2e18, 2))
+			.then((result) => {
+				sealedBid = result;
+				return registrar.newBidAsync(result, {from: accounts[1], value: 2e18});
+			})
+			.then((done) => advanceTimeAsync(days(3) + 1))
+			.then((done) => registrar.unsealBidAsync(web3.sha3('name'), 1e18, 1, {from: accounts[0]}))
+			.then((done) => registrar.unsealBidAsync(web3.sha3('name'), 2e18, 2, {from: accounts[1]}))
+			.then((done) => advanceTimeAsync(days(2) + 1))
+			.then((done) => registrar.finalizeAuctionAsync(web3.sha3('name'), {from: accounts[1]}))
+			.then((done) => advanceTimeAsync(days(366) + 1))
```

Comment:
Can you readd tests for releasing the deed early, and releasing the deed as the wrong user?